### DISCUSSION TITLE
ObservableProperty & NotifyCanExecuteChanged

### DISCUSSION THREAD

**Discussion by gazzamundo** [CommunityToolkit/dotnet#82] [Q&A] [answered]
I am declaring properties using the quick attribute tag '[ObservableProperty]', this is working great as a shorthand for declaring a property that implements that should be wrapped by a generated observable object. 
However, when some property values change, I need to inform a command that its execution status has changed, my question is, is there any shorthand way of doing this in conjunction with the [ObservableProperty] attribute, or, am I forced to remove the [ObservableProperty] attribute and go to a full property syntax with get/set and then within the set, use SomeCommand.NotifyCanExecuteChanged?

Many thank in advance

**Comment by redawgts** [comment]
There's a [AlsoNotifyCanExecuteFor] attribute in the source code. There hasn't been an updated NuGet package since before it was added though. I'm sure it will be in the next update. Until then your stuck using a full property.

**Reply by haltcase** [reply]
[8.0.0-preview2](https://www.nuget.org/packages/CommunityToolkit.Mvvm/8.0.0-preview2) has been released and appears to include `[AlsoNotifyCanExecuteFor]`, but there's no release notes quite yet.

**Reply by Sergio0694** [reply]
Ah, that's because I forgot to write them down, whoops 😅
I'll go do that, thanks!

**Comment by Sergio0694** [comment, marked as answer]
As others have mentioned, you can do this by using `[AlsoNotifyCanExecuteFor]`, which is available in 8.0.0 Preview releases 🙂

**Reply by golemme** [reply]
AlsoNotifyCanExecuteFor and the AllowConcurrentExecutions parameter for ICommand are missing from Preview Release 3 -- unless the currently available NuGet package (version 7.1.2) is actually not Preview Release 3. Will they be back in Release 4 (or will the NuGet package be updated)?

**Reply by Sergio0694** [reply]
Wait what? They're... Not missing? 🤔
I most definitely did not remove them, they should 100% still be there.

**Reply by haltcase** [reply]
@golemme I don't think you're on Preview Release 3 if you're using 7.1.2. The previews are in the 8.x line, so you'd be looking for [8.0.0-preview3](https://www.nuget.org/packages/CommunityToolkit.Mvvm/8.0.0-preview3).

**Reply by golemme** [reply]
8.0.0-preview3 doesn't appear in my NuGet Package Manager in Visual Studio 2022, and this is the first time I have ever seen the link that @haltcase provided. I assumed the CommunityToolkit.Mvvm that appears in the NuGet Package Manager is up-to-date (since James Montemagno installed it that way in a video in which he demonstrates the code generators that I am missing), but apparently that's not the case. Thanks for your help.

**Reply by Sergio0694** [reply]
@golemme You need to check the box for "preview releases", otherwise it won't show up.
